### ISSUE TITLE
Type error when importing using TypeScript

### ISSUE THREAD

**Issue by Zamiell** [ota-meshi/jsonc-eslint-parser#247]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [x] I'm using [eslint-plugin-jsonc](https://github.com/ota-meshi/eslint-plugin-jsonc). (`*.json` file linting does not work with the parser alone. You should also use eslint-plugin-jsonc with it.)
- [x] I'm sure the problem is a parser problem. (If you are not sure, search for the issue in [eslint-plugin-jsonc](https://github.com/ota-meshi/eslint-plugin-jsonc) repo and open the issue in [eslint-plugin-jsonc](https://github.com/ota-meshi/eslint-plugin-jsonc) repo if there is no solution.
- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.

### What version of ESLint are you using?

9.39.1

### What version of `eslint-plugin-jsonc` and `jsonc-eslint-parser` are you using?

- jsonc-eslint-parser@2.4.1
- eslint-plugin-jsonc@2.21.0

### What did you do?

```
// ts-check

import jsoncParser from "jsonc-eslint-parser";
```

### What did you expect to happen?

I expected to get no type errors from TypeScript.

### What actually happened?

I got the following error:

```
Module '"c:/Users/jnesta/Repositories/infrastructure/node_modules/jsonc-eslint-parser/lib/index"' has no default export. ts(1192)
```

### Link to **GitHub Repo** with Minimal Reproducible Example

```sh
mkdir test
cd test
npm init --yes
npm install --save typescript jsonc-eslint-parser
echo '
// @ts-check

import jsoncParser from "jsonc-eslint-parser";
' > eslint.config.mjs
npx tsc eslint.config.mjs --allowJs --noEmit --skipLibCheck
```

Will display the following error:

```
eslint.config.mjs:4:8 - error TS1192: Module '"C:/Users/jnesta/Repositories/test/node_modules/jsonc-eslint-parser/lib/index"' has no default export.

4 import jsoncParser from "jsonc-eslint-parser";
         ~~~~~~~~~~~


Found 1 error in eslint.config.mjs:4
```

Note that the error goes away if you change the import statement to be:

```js
import * as jsoncParser from "jsonc-eslint-parser";
```

However, if that is the intended way to import this parser, then the README.md should be changed.

One way to fix this bug is to change the type definitions file to say that a default export exists.

(However, I think [default exports are evil](https://blog.piotrnalepa.pl/2020/06/26/default-exports-vs-named-exports/) so I would actually prefer that the library be changed to have a named export instead (and enforce it using [`import-x/no-default-export`](https://github.com/un-ts/eslint-plugin-import-x/blob/master/docs/rules/no-default-export.md). But of course, its just my opinion, and reasonable minds may disagree. =p)

**Comment by ota-meshi** [comment]
This package is written in ESM but published in CommonJS.
Changes that are incompatible with CommonJS would be problematic, but changes that are compatible should be OK.

**Comment by ota-meshi** [comment]
Looking at your PR, I'm not sure what your issue is 😓 

Could you share a repository where we can see TS error issue?
Also, please make sure JSON Lint works on that repository.

**Comment by Zamiell** [comment]
> Could you share a repository where we can see TS error issue?

I already did in the OP! Just copy paste the text from the "Minimal Reproducible Example" into your terminal and you will see the error.

**Comment by ota-meshi** [comment]
Thanks to you, I now have all the information, so I no longer need the repository.
https://github.com/ota-meshi/jsonc-eslint-parser/pull/251#issuecomment-3615168218

I think using `import * as jsoncParser from "jsonc-eslint-parser";` is the way to go.
If you're interested, please open a PR to fix the documentation.
